### ISSUE TITLE
Merge Facilities from specific contribs via Script 

### ISSUE THREAD

**Issue by mariel-oar** [open-apparel-registry/open-apparel-registry#1782]
Two common data moderation activities on the OAR are facility merges and promotion of data points from specific contributors. 

At some point leading up to delivery of OGR 1.0 (MVP / July 1) we would like to execute a bulk merge and bulk data point promotion (if possible) via scripts. An past example of this activity is when facilities identified as duplicative by Isaac were batch merged into their target facilities. 

The OAR team's understanding of what is required to make this happen is: 

- Target merge facilities and facility(ies) that should be merged into the target need to be identified (required information for this to occur is the match ID) 
- Data points that should be promoted on a given facility profile need to be identified (Q @hlennett, which information / id is required to identify the data point to be promoted?) 

---

The tools/mergeduplicatefacilities.py script was added in #1477

**Comment by hlennett** [comment]
@mariel-oar Timing update on this: Each of the team members prepping the data will plan to have their data ready by the 18th. I want a day to check them/put them all together after that, so we can plan to hand off to Azavea on the 20th, if that fits in well with their sprint.

**Comment by mariel-oar** [comment]
OK, thanks @hlennett. April 20 will be the second week of Azavea's sprint. Given other work they they have in the [priority list ](https://app.zenhub.com/workspaces/civic-apps-5c8fabc1177a1c1f6672f901/board?repos=159699912,283246038), this may push to the next sprint. 

I'm thinking this either belongs after #1767 or #1788, depending on how urgently it's needed (i think after 1788 may mean it gets pushed out to the sprint after next). What do you think?

**Comment by hlennett** [comment]
I suppose I'm hoping that because it's a 1 point issue that they've already written the script for, it is hopefully quite easy to slot in? From a trust in data/data quality perspective, the sooner we get it in, the better. I'd put it after 1781.

**Comment by hlennett** [comment]
Here is the data for the script run: https://docs.google.com/spreadsheets/d/1BucsaNh79JjxQpm428PJHD1VpBVTej92ekXqbK29ztI/edit?usp=sharing

**Comment by emilyhu0106** [comment]
Hi @mariel-oar and @hlennett! The facilities in the spreadsheet are merged now, please take a look when you have time. We have merging error from 9 out of ~600 facilities (~1.5%), mostly because of the facility ID doesn't exist. The detailed message is below:
```
./tools/mergeduplicatefacilitiescsv.py 2022_Duplicate_script.csv -t [TOKEN] -r https://openapparel.org
('Status:', 400, 'Content:', '{"merge":["Facility IN2022010BQQECH does not exist."]}', 'Target ID:', 'IN2020078X5J8DK', 'Merge ID:', 'IN2022010BQQECH')
('Status:', 400, 'Content:', '{"target":["Facility IN2020084JHR7GS does not exist."]}', 'Target ID:', 'IN2020084JHR7GS', 'Merge ID:', 'IN202120195YD91')
('Status:', 400, 'Content:', '{"target":["Facility MM2020268JJ8Q2K does not exist."]}', 'Target ID:', 'MM2020268JJ8Q2K', 'Merge ID:', 'MM2019083TMN7JD')
('Status:', 400, 'Content:', '{"target":["Facility MA2021077QXJKC6 does not exist."]}', 'Target ID:', 'MA2021077QXJKC6', 'Merge ID:', 'MA2021270MDAHSH')
('Status:', 400, 'Content:', '{"target":["Facility MA2021077QXJKC6 does not exist."]}', 'Target ID:', 'MA2021077QXJKC6', 'Merge ID:', 'MA2022082T9HMDK')
('Status:', 400, 'Content:', '{"merge":["Facility MM2019158XS9VV9 does not exist."]}', 'Target ID:', 'MM2019108WW2B9X', 'Merge ID:', 'MM2019158XS9VV9')
('Status:', 400, 'Content:', '{"merge":["Facility MM2021175Q2XZJK does not exist."]}', 'Target ID:', 'MM20203640BGYKQ', 'Merge ID:', 'MM2021175Q2XZJK')
('Status:', 400, 'Content:', '{"merge":["Facility LK2021342QES1VZ does not exist."]}', 'Target ID:', 'LK2019098Z3ERHD', 'Merge ID:', 'LK2021342QES1VZ')
('Status:', 400, 'Content:', '{"merge":["Facility LK2019087H404CQ does not exist."]}', 'Target ID:', 'LK2019098Z3ERHD', 'Merge ID:', 'LK2019087H404CQ')
```

Thank you!

**Comment by vrwOAR** [comment]
Hi Emily - thanks for this update. 

I reviewed the error list and all except for two have been removed. They got moderated by OAR team before the script ran. These two remain:

- MO6;MA2022082T9HMDK;G C Morocco ltd;4 Avenue Othman Ibn Affan Z.I Tabriquet, Sale, Morocco - Morocco
- MO6;MA2021270MDAHSH;G C Morocco Sarl;Zone Industrielle Sale Tabriquet Lot N°4 Avenue Othmane Ibn Afane  Sale

**Comment by emilyhu0106** [comment]
Hi-

I checked these two facilities you mentioned and found that `MA2021270MDAHSH` is merged with `MA2022082T9HMDK`. 

Do you mind confirming this on your end or provide more context here? Thank you!

**Comment by vrwOAR** [comment]
Hi @emilyhu0106 the merge is complete. I entered it manually. We're all set on this ticket. Thanks for checking in.

Vanessa
